Assets/Scripts/Enemy/EnemyFactory.cs

```diff
@@ -39,33 +46,41 @@ public static EnemyFactory Instance
 
     public EnemyFactory()
     {
+        // Define enemyInfo for each type.
+        _enemyInfo.Add( EnemyType.TURRET, new EnemyInfo(_defaultFunc, Globals.DirectTargeting, _defaultMove));
+        
         // Set up the dictionary with methods.
-        _enemySetups.Add("Turret", CreateTurret);
+        _enemyPostSetups.Add(EnemyType.TURRET, CreateTurret);
     }
     
-    // TODO: Determine how we want our targeting, moving, and pathfinding to be passed in or set up.
-    public GameObject CreateEnemy(string enemyName, Vector3 position)
+    public GameObject CreateEnemy(EnemyType enemyType, Vector3 position)
     {
         // Janky, find a better place for this. Unsure when the Player object is available.
         if (_defaultTarget == null) _defaultTarget = GameObject.FindGameObjectWithTag("Player");
+
+        if (!Globals.enemyPrefabNames.ContainsKey(enemyType)) return null;
+        
+        string enemyName = Globals.enemyPrefabNames[enemyType];
         
-        // We expect enemyName to be the name of the resource.
         GameObject newEnemyObject = GameObject.Instantiate(Resources.Load(enemyName)) as GameObject;
         Transform enemyTransform = newEnemyObject.GetComponent<Transform>();
         
         // TODO: change default vector to dynamically adjust height of enemy spawn so they don't spawn under the ground.
         enemyTransform.position = position + new Vector3(0, 1, 0);
         
-        // TODO: Determine whether we're including EnemyBehvaviour in the prefab, or adding it as component.
-        // I'll just go with adding it for now.
+        Weapon enemyWeapon = newEnemyObject.AddComponent<Weapon>();

```

Comment:
I'm not sure why GitHub thinks this is a new addition, this is from another PR that was already merged.